fvm/handler/accountKey_test.go

```diff
@@ -0,0 +1,101 @@
+package handler
+
+import (
+	"testing"
+	"unicode/utf8"
+
+	"github.com/fxamacker/cbor/v2"
+	"github.com/onflow/cadence"
+	"github.com/onflow/cadence/runtime"
+	"github.com/onflow/flow-go/crypto"
+	"github.com/onflow/flow-go/crypto/hash"
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestAddEncodedAccountKey_error_handling_produces_valid_utf8(t *testing.T) {
+
+	akh := &AccountKeyHandler{accounts: FakeAccounts{}}
+
+	address := cadence.BytesToAddress([]byte{1, 2, 3, 4})
+
+	// emulate encoded public key (which comes as a user input)
+	// containing bytes which are invalid UTF8
+
+	invalidEncodedKey := make([]byte, 64)
+	invalidUTF8 := []byte{0xc3, 0x28}
+	copy(invalidUTF8, invalidEncodedKey)
+	publicKey := FakePublicKey{data: invalidEncodedKey}
+
+	accountPublicKey := flow.AccountPublicKey{
+		Index:     1,
+		PublicKey: publicKey,
+		SignAlgo:  crypto.ECDSASecp256k1,
+		HashAlgo:  hash.SHA3_256,
+		SeqNumber: 0,
+		Weight:    1000,
+		Revoked:   false,
+	}
+
+	encodedPublicKey, err := flow.EncodeRuntimeAccountPublicKey(accountPublicKey)
+	require.NoError(t, err)
+
+	err = akh.AddEncodedAccountKey(runtime.Address(address), encodedPublicKey)
+	require.Error(t, err)
```

Comment:
lets check the specific error type here, you are expecting to return

Reply:
Done and done